### DISCUSSION TITLE
Getting Authorization policy missing or undefined Error

### DISCUSSION THREAD

**Discussion by nitheshrayuduv** [open-policy-agent/community#341] [Community]
If I start the OPA application in windows using the below run command. 

opa run --server --log-level debug --tls-cert-file public.crt --tls-private-key-file private.key --authentication=token --authorization=basic

Always ending up with, Is there any other API endpoint and REGO policy i need to add, and how to add it ? 

{
    "code": "internal_error",
    "message": "authorization policy missing or undefined"
}

**Comment by anderseknert** [comment]
Hi @nitheshrayuduv 👋 Have you provided an authorization policy for OPA? The process is described [here](https://www.openpolicyagent.org/docs/latest/security/#authentication-and-authorization).

**Reply by nitheshrayuduv** [reply]
Hi @anderseknert , thank you for the information, 

1) tried it and still getting the same error.

  PUT with https://localhost:8181/v1/policies/nithpolicy
  
  for the below policy (authz.rego)
  
  package system.authz
  
  default allow := false           # Reject requests by default.
  
  allow {                         # Allow request if...
      "secret" == input.identity  # Identity is the secret root key.
  }

2) Tried to execute the below cur command also didnt work.

curl -X PUT --data-binary @policies/authz.rego https://localhost:8181/authorize/v1/policies/httpapi/authz

Am i missing the correct URL or the policy.

**Reply by anderseknert** [reply]
You won't be able to provide the authorization policy via PUT, since the server requires authorization :) You'll need to either provide it via a bundle, or use the simpler option — just provide the policy file for the `opa run` command directly, i.e.

```
opa run --server --log-level debug --tls-cert-file public.crt --tls-private-key-file private.key --authentication=token --authorization=basic authz.rego
```

**Reply by nitheshrayuduv** [reply]
Thanks @anderseknert , Yes worked now tried with this opa run file:///C:/soft/OPA/policies/authz.rego --server --authentication=token --authorization=basic, where the file is the base authorization rego file.
